src/cli/cli_srp_client.cpp

```diff
@@ -0,0 +1,486 @@
+            otIp6Address addresses[kMaxHostAddresses];
+
+            numAddresses = 0;
+            memset(addresses, 0, sizeof(addresses));
+
+            while (aArgsLength > 1)
+            {
+                VerifyOrExit(numAddresses < kMaxHostAddresses, error = OT_ERROR_NO_BUFS);
+                aArgsLength--;
+                aArgs++;
+                SuccessOrExit(error = ParseAsIp6Address(aArgs[0], addresses[numAddresses++]));
+            }
+
+            VerifyOrExit(numAddresses > 0, error = OT_ERROR_INVALID_ARGS);
+            memcpy(mHostAddresses, addresses, sizeof(addresses));
+
+            error = otSrpClientSetHostAddresses(mInterpreter.mInstance, mHostAddresses, numAddresses);
+        }
+    }
+    else if (strcmp(aArgs[0], "remove") == 0)
+    {
+        bool removeKeyLease = false;
+
+        if (aArgsLength > 1)
+        {
+            VerifyOrExit(aArgsLength == 2, error = OT_ERROR_INVALID_ARGS);
+            SuccessOrExit(error = ParseAsBool(aArgs[1], removeKeyLease));
+        }
+
+        error = otSrpClientRemoveHostAndServices(mInterpreter.mInstance, removeKeyLease);
+    }
+    else if (strcmp(aArgs[0], "clear") == 0)
+    {
+        VerifyOrExit(aArgsLength == 1, error = OT_ERROR_INVALID_ARGS);
+        otSrpClientClearHostAndServices(mInterpreter.mInstance);
+
+        for (Service &poolEntry : mServicePool)
+        {
+            poolEntry.MarkAsNotInUse();
+        }
+    }
+    else
+    {
+        error = OT_ERROR_INVALID_COMMAND;
+    }
+
+exit:
+    return error;
+}
+
+otError SrpClient::ProcessLeaseInterval(uint8_t aArgsLength, char *aArgs[])
+{
+    otError  error = OT_ERROR_NONE;
+    uint32_t interval;
+
+    if (aArgsLength == 0)
+    {
+        mInterpreter.OutputLine("%u", otSrpClientGetLeaseInterval(mInterpreter.mInstance));
+        ExitNow();
+    }
+
+    VerifyOrExit(aArgsLength == 1, error = OT_ERROR_INVALID_ARGS);
+    SuccessOrExit(error = ParseAsUint32(aArgs[0], interval));
+    otSrpClientSetLeaseInterval(mInterpreter.mInstance, interval);
+
+exit:
+    return error;
+}
+
+otError SrpClient::ProcessKeyLeaseInterval(uint8_t aArgsLength, char *aArgs[])
+{
+    otError  error = OT_ERROR_NONE;
+    uint32_t interval;
+
+    if (aArgsLength == 0)
+    {
+        mInterpreter.OutputLine("%u", otSrpClientGetKeyLeaseInterval(mInterpreter.mInstance));
+        ExitNow();
+    }
+
+    VerifyOrExit(aArgsLength == 1, error = OT_ERROR_INVALID_ARGS);
+    SuccessOrExit(error = ParseAsUint32(aArgs[0], interval));
+    otSrpClientSetKeyLeaseInterval(mInterpreter.mInstance, interval);
+
+exit:
+    return error;
+}
+
+otError SrpClient::ProcessService(uint8_t aArgsLength, char *aArgs[])
+{
+    otError error = OT_ERROR_NONE;
+
+    if (aArgsLength == 0)
+    {
+        OutputServiceList(0, otSrpClientGetServices(mInterpreter.mInstance));
+        ExitNow();
+    }
+
+    if (strcmp(aArgs[0], "add") == 0)
+    {
+        // `add` <instance-name> <service-name> <port> [priority] [weight]
```

Comment:
How about adding another argument `[<TXT>]`, we see TXT related code on both client & service side is not covered in https://github.com/openthread/openthread/pull/6074.

Reply:
Added `[txt`] parameter to the CLI command. 
It is expected as a hex string byte sequence (already encoded).